### PR TITLE
Remove unnecessary OpenSSL 1.1.1 matrix, push branch jobs

### PR DESCRIPTION
**Pull request by sethmlarson**
Maintain the Python 3.8 + OpenSSL job, drop the rest as a waste of CI resources.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -38,11 +42,9 @@ jobs:
         os:
           - macos-11
           - windows-latest
-          - ubuntu-20.04  # OpenSSL 1.1.1
-          - ubuntu-22.04  # OpenSSL 3.0
+          - ubuntu-22.04
         nox-session: ['']
         include:
-          - experimental: false
```
Comment (sethmlarson) [inline .github/workflows/ci.yml:45]: This entry looked like a mistake? Not sure.

**Review by pquentin — DISMISSED** [review]
Thanks! LGTM. I'd like to know what @illia-v thinks too.

**Comment by sethmlarson** [comment]
Thanks for the review, I'll wait for @illia-v to merge :)

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -1,6 +1,10 @@
 name: CI
 
-on: [push, pull_request]
+on:
+  pull_request:
+  push:
+    branches:
+      - main
```
[the remark this answers is not in the document]
Comment (illia-v) [reply .github/workflows/ci.yml:7]: The feature of testing in my fork was useful for me a few times, but it has been excessive most of the time. If this is changed here, can we add `workflow_dispatch` instead to be able to dispatch tests manually if needed?

[the remark this answers is not in the document]
Comment (sethmlarson) [reply .github/workflows/ci.yml:7]: Maybe I can revert this part of the PR, if both of you are using it I think others likely are too! :)

**Review by illia-v — APPROVED** [review]
Thanks!

**Pull request merged**